File: Applied Course/4.Problem Solving/Leetcode Callenge/2.Move Zeros.py
# https://leetcode.com/problems/move-zeroes/
# Logic-1
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        result=[]
        for num in nums:
            if num!=0:
                result.append(num)
        for num in nums:
            if num==0:
                result.append(num)
        for i in range(len(nums)):
            nums[i]=result[i]
# Logic-2
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n=len(nums)
        last_non_zero_index=0
        for i in range(n):
            if nums[i]!=0:
                nums[last_non_zero_index]=nums[i]
                last_non_zero_index+=1
        for i in range(last_non_zero_index,n):
            nums[i]=0

File: Applied Course/4.Problem Solving/Leetcode Callenge/2.Move Zeros.cpp
#include<bits/stdc++.h>
using namespace std;
// Logic-1
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        vector<int> result;
        for(auto num:nums)
            if(num!=0)
                result.push_back(num);
        for(auto num:nums)
            if(num==0)
                result.push_back(num);
        nums=result;
    }
};
// Logic-2
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n=nums.size();
        int last_non_zero_index=0;
        for(int i=0;i<n;i++)
        {
            if(nums[i]!=0)
            {
                nums[last_non_zero_index]=nums[i];
                last_non_zero_index+=1;
            }
        }
        for(int i=last_non_zero_index;i<n;i++)
            nums[i]=0;
    }
};
int main()
{
    // https://leetcode.com/problems/move-zeroes/
    return 0;
}